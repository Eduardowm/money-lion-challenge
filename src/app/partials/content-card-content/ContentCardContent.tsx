'use client'

import Button from '@/components/ui/button/Button'
import { ContentCard } from '@/data/entities/ContentCardEntity'
import { cn } from '@/lib/helpers/tailwind'
import { useState } from 'react'

interface ContentCardContentProps {
  post: ContentCard
}

function ContentCardContent({ post }: ContentCardContentProps) {
  const [showFullContent, setShowFullContent] = useState<boolean>(false)

  return (
    <div className="flex flex-col px-1">
      <div className="flex w-full justify-between">
        <div className="flex flex-col">
          <h2 className="text-base sm:text-lg font-semibold">{post.title}</h2>
          <p className="text-sm sm:text-base text-gray-400">{post.subtitle}</p>
        </div>
        {post.author && (
          <span className="text-gray-400 text-nowrap w-fit text-sm sm:text-base">{post.author.name}</span>
        )}
      </div>

      <div className={cn('w-full text-sm sm:text-base mt-2', !showFullContent && 'text-lines-limit-3')}>
        {post.content}
      </div>

      {!showFullContent && (
        <div className="flex justify-end">
          <Button className="text-right text-sm sm:text-base" onClick={() => setShowFullContent(true)}>
            Read more
          </Button>
        </div>
      )}
    </div>
  )
}
export default ContentCardContent
