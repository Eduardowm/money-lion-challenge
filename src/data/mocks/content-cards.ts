import { ContentCard } from '../entities/ContentCardEntity'

export const mockRawContentCard = {
  id: 'abc-def-ghi',
  imageUri: '/images/mock.jpg',
  textData: {
    title: 'A mock title',
    subTitle: 'A mock subtitle',
    body: 'A mock content with a lot of text to fill the space',
    author: { first: 'Mock', last: 'author' },
  },
  metadata: {
    priority: 1,
    publishDate: '2021-01-01',
  },
  comments: [
    {
      text: 'A mock comment',
      author: 'Commenter 1',
      profilePic: '/images/mock.jpg',
      likes: 20,
    },
    {
      text: 'The second mock comment',
      author: 'Commenter 2',
      profilePic: '/images/mock2.jpg',
      likes: 50,
    },
  ],
}

export const mockContentCards: ContentCard[] = [
  {
    id: 'abc-def-ghi',
    title: 'A mock title',
    subtitle: 'A mock subtitle',
    content: 'A mock content with a lot of text to fill the space',
    priority: 1,
    author: {
      name: 'Mock author',
    },
    image: '/images/mock.jpg',
    comments: [
      {
        name: 'Commenter 1',
        image: '/images/mock.jpg',
        text: 'A mock comment',
        likes: 20,
      },
      {
        name: 'Commenter 2',
        image: '/images/mock2.jpg',
        text: 'The second mock comment',
        likes: 50,
      },
    ],
  },
  {
    id: 'ihg-fed-cba',
    title: 'An alternative mock title',
    subtitle: 'An alternative mock subtitle',
    content: 'An alternative mock content with a lot of text to fill the space',
    priority: 1,
    author: {
      name: 'An alternative mock author',
    },
    image: '/images/mock.jpg',
    comments: [
      {
        name: 'Alternative commenter 1',
        image: '/images/mock.jpg',
        text: 'An alternative mock comment',
        likes: 20,
      },
    ],
  },
]
