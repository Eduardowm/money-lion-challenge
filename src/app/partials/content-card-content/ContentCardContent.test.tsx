import { render } from '@testing-library/react'

import { mockContentCards } from '@/data/mocks/content-cards'
import ContentCardContent from './ContentCardContent'

describe('ContentCardContent', () => {
  it('renders as expected', () => {
    const { container } = render(<ContentCardContent post={mockContentCards[0]} />)

    expect(container).toMatchSnapshot()
  })
})
