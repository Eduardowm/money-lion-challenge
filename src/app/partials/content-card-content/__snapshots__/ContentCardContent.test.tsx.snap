// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentCardContent renders as expected 1`] = `
<div>
  <div
    class="flex flex-col px-1"
  >
    <div
      class="flex w-full justify-between"
    >
      <div
        class="flex flex-col"
      >
        <h2
          class="text-base sm:text-lg font-semibold"
        >
          A mock title
        </h2>
        <p
          class="text-sm sm:text-base text-gray-400"
        >
          A mock subtitle
        </p>
      </div>
      <span
        class="text-gray-400 text-nowrap w-fit text-sm sm:text-base"
      >
        Mock author
      </span>
    </div>
    <div
      class="w-full text-sm sm:text-base mt-2 text-lines-limit-3"
    >
      A mock content with a lot of text to fill the space
    </div>
    <div
      class="flex justify-end"
    >
      <button
        class="text-gray-500 hover:text-gray-300 text-right text-sm sm:text-base"
      >
        Read more
      </button>
    </div>
  </div>
</div>
`;
